Source/Csla.Blazor/ViewModel.cs

```diff
@@ -236,15 +237,11 @@ public async Task SaveAsync()
       {
         if (Model is Core.ITrackStatus obj && !obj.IsSavable)
         {
-          if (obj.IsBusy)
+          if (obj is INotifyBusy notifyBusy)
           {
-            var stopTime = DateTime.Now + BusyTimeout;
-            while (obj.IsBusy)
-            {
-              if (DateTime.Now > stopTime)
-                throw new TimeoutException("SaveAsync");
-              await Task.Delay(1);
-            }
+            var cslaOptions = ApplicationContext.GetRequiredService<Csla.Configuration.CslaOptions>();
+            var _waitForIdleTimeout = TimeSpan.FromSeconds(cslaOptions.DefaultWaitForIdleTimeoutInSeconds);
+            await BusyHelper.WaitForIdle(notifyBusy, _waitForIdleTimeout, "SaveAsync").ConfigureAwait(false);
```

Comment:
"SaveAsync" as parameter is unnecessary. The method uses `CallerMemberName` to get that.

Reply:
ok. I will update that.